File: ChatGPT/Java/Facade/round2/source/CarFacade.java
package ChatGPT.Java.Facade.round2.source;

public class CarFacade {
     private Engine engine;
     private Lights lights;
     private Steering steering;
     private Brakes brakes;
 
     public CarFacade() {
         this.engine = new Engine();
         this.lights = new Lights();
         this.steering = new Steering();
         this.brakes = new Brakes();
     }
 
     public void startCar() {
         engine.startEngine();
         lights.turnOnHeadlights();
     }
 
     public void stopCar() {
         lights.turnOffHeadlights();
         engine.stopEngine();
     }
 
     public void turnLeft() {
         steering.turnLeft();
     }
 
     public void turnRight() {
         steering.turnRight();
     }
 
     public void applyBrakes() {
         brakes.applyBrakes();
     }
 
     public void releaseBrakes() {
         brakes.releaseBrakes();
     }
 }
 

File: ChatGPT/Python/Facade/round2/source/car_facade.py
# car_facade.py
from engine import Engine
from lights import Lights
from steering import Steering
from brakes import Brakes

class CarFacade:
    def __init__(self):
        self.engine = Engine()
        self.lights = Lights()
        self.steering = Steering()
        self.brakes = Brakes()

    def start_car(self):
        self.engine.start_engine()
        self.lights.turn_on_headlights()

    def stop_car(self):
        self.lights.turn_off_headlights()
        self.engine.stop_engine()

    def turn_left(self):
        self.steering.turn_left()

    def turn_right(self):
        self.steering.turn_right()

    def apply_brakes(self):
        self.brakes.apply_brakes()

    def release_brakes(self):
        self.brakes.release_brakes()